src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -108,10 +108,28 @@ private static void createReactivePowerControlBranchEquation(LfBranch branch, Re
     }
 
     private static void createShuntEquations(LfBus bus, EquationSystem<AcVariableType, AcEquationType> equationSystem, VariableSet<AcVariableType> variableSet) {
-        for (LfShunt shunt : bus.getShunts()) {
-            ShuntCompensatorReactiveFlowEquationTerm q = new ShuntCompensatorReactiveFlowEquationTerm(shunt, bus, variableSet);
-            equationSystem.createEquation(bus.getNum(), AcEquationType.BUS_Q).addTerm(q);
-            shunt.setQ(q);
+        List<LfShunt> controllerShunts = bus.getControllerShunts();
+        if (controllerShunts.isEmpty()) {
+            for (LfShunt shunt : bus.getShunts()) {
+                ShuntCompensatorReactiveFlowEquationTerm q = new ShuntCompensatorReactiveFlowEquationTerm(shunt, bus, variableSet, false);
+                equationSystem.createEquation(bus.getNum(), AcEquationType.BUS_Q).addTerm(q);
+                shunt.setQ(q);
+            }
+        } else {
+            Optional<DiscreteVoltageControl> dvc = bus.getDiscreteVoltageControl();
+            if (dvc.isPresent() && dvc.get().getControllerBuses().contains(bus)) {
```

Comment:
This part is the most hard to understand and probably the implementation is not the good one. In case of several shunts, instead of creating several B variables, I have decided to create just one variable B that takes all we need to control voltage and to redispatch the amount of B on the physical shunts after the resolution.

Reply:
You have to use only one b variable, but maybe you could improve your implementation by spreading the b into each shunt by just adding a scaler to the term.

Reply:
Can you explain with more details what you have in mind ? In my implementation, I use only one `b` variable on a bus, but if I have several shunts, I have to find the best combination of the sections of the shunts that is close to the solved `b`. How can we do that through a scalar ?

Reply:
So we cannot just have a proportional distribution of b across all shunts directly in the equation system? 
I don't remember why we create as many LfShunt by bus as in the IIDM network. Maybe it would be simpler to have only one LfShunt by LfBus that would aggregate IIDM shunt compensator. In the case of non regulating shunt compensators that would only required in the updateState method to distribute the q proportionally to individual b and in your regulating shunt case that would be simpler to model.

Reply:
The distribution could not be proportional to be optimal because each shunt have sections (in a different number too), that could be non linear too. So I have to fill the shunt one by one to find the optimal combinaison that is the closest to the continuous value. But for sure, it would be easier to have a single LfShunt by a bus, with one or two shunt equations: one for the constant B shunts and the other one for the regulating ones. But it is a huge change... and maybe the refactoring should be done before or after this PR.